src/daemon/default_vm_image_vault.cpp

```diff
@@ -605,6 +632,22 @@ mp::VMImage mp::DefaultVMImageVault::fetch_kernel_and_initrd(const VMImageInfo&
     return image;
 }
 
+mp::VMImage mp::DefaultVMImageVault::finalize_image_records(const Query& query, const VMImage& prepared_image)
+{
+    VMImage vm_image;
+
+    if (!query.name.empty())
```

Comment:
Ok, I'll see if this can be dropped altogether.

Reply:
After looking at this, `query.name` can indeed be empty, such as when the background image updater runs.  That only downloads new images and doesn't prepare any instance images.  So no, it can't be dropped and I prefer to keep it here as well.

Reply:
I see, alright then

Reply:
Can you add a comment then? As that's not obvious to me either